DigitalLearningSolutions.Data/Helpers/FilteringHelper.cs

```diff
@@ -39,19 +39,20 @@ public static string BuildFilterValueString(string group, string propertyName, s
         public static string? GetFilterString(
             string? existingFilterString,
             string? newFilterToAdd,
+            bool clearFilters,
             HttpRequest request,
             string cookieName,
             string? defaultFilterValue = null
         )
         {
-            if (existingFilterString == null && newFilterToAdd == null)
+            if (existingFilterString == null && newFilterToAdd == null && !clearFilters)
             {
                 return request.Cookies.ContainsKey(cookieName)
-                    ? request.Cookies[cookieName]
+                    ? request.Cookies[cookieName] == EmptyFiltersCookieValue ? null : request.Cookies[cookieName]
                     : defaultFilterValue;
             }
 
-            if (existingFilterString?.ToUpper() == ClearString)
+            if (clearFilters)
             {
                 existingFilterString = null;

```

Comment:
If we get given a newFilterToAdd in this method alongside our `clearFilters` flag then we'll end up setting the newFilterToAdd right? We don't ever want to do that right now, and neither is it possible for that to happen. I think we should return in this if statement.
The bonus of doing that is that you can make the logic a bit clearer by having 
`if (clearFitlers)... else if (filter not provided by query params)... else ...`